python/tk_maya/menu_generation.py

```diff
@@ -294,6 +294,20 @@ def add_command_to_menu(self, menu):
             
         pm.menuItem(**params)
         
+    def __exception_trap_callback_wrapper(self, callback):
+        """
+        Wrapper which traps exceptions. We execute menu commands
+        in maya via evaldeferred, meaning that all errors are just
+        swallowed by maya never to resurface. This wrapper prints 
+        them out.
+        """
+        try:
+            callback()
+        except Exception, e:
+            current_engine = tank.platform.current_engine()
```

Comment:
i think it's possible if there is a menu which is "dangling" after an engine restart. But I really was in two minds about that if myself, I agree that it was probably overly cautious to have that check, so i removed it. I think if you end up in a scenario where you have the menu still present but no engine, there will be many other issues, not just this one, so fair to make this assumption.